assert fail throws Boxed Error

When this code is executed I get a Boxed Error, which can be confusing
```scala
class Test extends munit.FunSuite {

  case class Inner(value: Double)

  case class Outer(data: Seq[Inner])

  test("exist issue") {
    val values = Array(
      Outer(Seq(Inner(1), Inner(2), Inner(10))),
      Outer(Seq(Inner(1), Inner(2), Inner(10))),
      Outer(Seq(Inner(1), Inner(2), Inner(10)))
    )
    assert(values.exists(outer => outer.data.exists(inner => inner.value > 90)))
  }

}
```
Result:
```
java.util.concurrent.ExecutionException: Boxed Error

    at scala.concurrent.impl.Promise$.resolver(Promise.scala:87)
    at scala.concurrent.impl.Promise$.scala$concurrent$impl$Promise$$resolveTry(Promise.scala:79)
    at scala.concurrent.impl.Promise$KeptPromise$.apply(Promise.scala:406)
    at scala.concurrent.Promise$.fromTry(Promise.scala:142)
    at scala.concurrent.Future$.fromTry(Future.scala:640)
    at munit.ValueTransforms.munitValueTransform(ValueTransforms.scala:37)
    at munit.ValueTransforms.munitValueTransform$(ValueTransforms.scala:22)
    at munit.FunSuite.munitValueTransform(FunSuite.scala:11)
    at munit.FunSuite.$anonfun$test$1(FunSuite.scala:37)
    at munit.MUnitRunner.$anonfun$runTestBody$1(MUnitRunner.scala:290)
Caused by: munit.FailException: /home/ale/workspace/munit_issue/src/test/scala/Test.scala:13 assertion failed
12:    )
13:    assert(values.exists(outer => outer.data.exists(inner => inner.value > 90)))
14:  }
    at munit.FunSuite.assert(FunSuite.scala:11)
    at Test.$anonfun$new$1(Test.scala:13)
    at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
```
Expected:
```
munit.FailException: /home/ale/workspace/munit_issue/src/test/scala/Test.scala:13 assertion failed
12:    )
13:    assert(values.exists(outer => outer.data.exists(inner => inner.value > 90)))
14:  }

    at munit.FunSuite.assert(FunSuite.scala:11)
    at Test.$anonfun$new$1(Test.scala:13)
    at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
```
If the assertion return true everything works as expected
munit version: 0.7.19
scala version: 2.11.12, 2.12.12

scala version 2.13.4 works as expected

--- a/munit/js/src/main/scala/java/lang/reflect/InvocationTargetException.scala
+++ b/munit/js/src/main/scala/java/lang/reflect/InvocationTargetException.scala
@@ -1,0 +1,3 @@
+package java.lang.reflect
+
+class InvocationTargetException extends java.lang.RuntimeException

--- a/munit/js/src/main/scala/java/lang/reflect/UndeclaredThrowableException.scala
+++ b/munit/js/src/main/scala/java/lang/reflect/UndeclaredThrowableException.scala
@@ -1,0 +1,3 @@
+package java.lang.reflect
+
+class UndeclaredThrowableException extends java.lang.RuntimeException

--- a/munit/shared/src/main/scala/munit/MUnitRunner.scala
+++ b/munit/shared/src/main/scala/munit/MUnitRunner.scala
@@ -4,7 +4,9 @@ import munit.internal.junitinterface.Configurable
 import munit.internal.PlatformCompat
 import org.junit.runner.Description
 import org.junit.runner.notification.RunNotifier
+import java.lang.reflect.InvocationTargetException
 import java.lang.reflect.Modifier
+import java.lang.reflect.UndeclaredThrowableException
 
 import munit.internal.FutureCompat._
 import munit.internal.console.StackTraces
@@ -245,19 +247,7 @@ class MUnitRunner(val cls: Class[_ <: Suite], newInstance: () => Suite)
         Future.successful(())
       case NonFatal(ex) =>
         trimStackTrace(ex)
-        val cause = ex match {
-          case e: ExecutionException if e.getCause() != null && {
-                e.getMessage() match {
-                  // NOTE(olafur): these exceptions appear when we await on
-                  // futures. We unwrap these exception in order to provide more
-                  // helpful error messages.
-                  case "Boxed Exception" | "Boxed Error" => true
-                  case _                                 => false
-                }
-              } =>
-            e.getCause()
-          case e => e
-        }
+        val cause = rootCause(ex)
         val failure = new Failure(description, cause)
         cause match {
           case _: AssumptionViolatedException =>
@@ -277,6 +267,16 @@ class MUnitRunner(val cls: Class[_ <: Suite], newInstance: () => Suite)
       notifier.fireTestFinished(description)
       true
     }
+  }
+
+  // NOTE(olafur): these exceptions appear when we await on futures. We unwrap
+  // these exception in order to provide more helpful error messages.
+  private def rootCause(x: Throwable): Throwable = x match {
+    case _: InvocationTargetException | _: ExceptionInInitializerError |
+        _: UndeclaredThrowableException | _: ExecutionException
+        if x.getCause != null =>
+      rootCause(x.getCause)
+    case _ => x
   }
 
   private def futureFromAny(any: Any): Future[Any] = any match {

--- a/munit/shared/src/main/scala/munit/MUnitRunner.scala
+++ b/munit/shared/src/main/scala/munit/MUnitRunner.scala
@@ -246,9 +246,15 @@ class MUnitRunner(val cls: Class[_ <: Suite], newInstance: () => Suite)
       case NonFatal(ex) =>
         trimStackTrace(ex)
         val cause = ex match {
-          case e: ExecutionException
-              if "Boxed Exception" == e.getMessage() &&
-                e.getCause() != null =>
+          case e: ExecutionException if e.getCause() != null && {
+                e.getMessage() match {
+                  // NOTE(olafur): these exceptions appear when we await on
+                  // futures. We unwrap these exception in order to provide more
+                  // helpful error messages.
+                  case "Boxed Exception" | "Boxed Error" => true
+                  case _                                 => false
+                }
+              } =>
             e.getCause()
           case e => e
         }

--- a/tests/shared/src/main/scala/munit/BoxedFrameworkSuite.scala
+++ b/tests/shared/src/main/scala/munit/BoxedFrameworkSuite.scala
@@ -1,0 +1,28 @@
+package munit
+
+class BoxedFrameworkSuite extends FunSuite {
+
+  case class Inner(value: Double)
+
+  case class Outer(data: Seq[Inner])
+
+  test("exist issue") {
+    val values = Array(
+      Outer(Seq(Inner(1), Inner(2), Inner(10))),
+      Outer(Seq(Inner(1), Inner(2), Inner(10))),
+      Outer(Seq(Inner(1), Inner(2), Inner(10)))
+    )
+    assert(values.exists(outer => outer.data.exists(inner => inner.value > 90)))
+  }
+
+}
+
+object BoxedFrameworkSuite
+    extends FrameworkTest(
+      classOf[BoxedFrameworkSuite],
+      """|==> failure munit.BoxedFrameworkSuite.exist issue - /scala/munit/BoxedFrameworkSuite.scala:15 assertion failed
+         |14:    )
+         |15:    assert(values.exists(outer => outer.data.exists(inner => inner.value > 90)))
+         |16:  }
+         |""".stripMargin
+    )

--- a/tests/shared/src/test/scala/munit/FrameworkSuite.scala
+++ b/tests/shared/src/test/scala/munit/FrameworkSuite.scala
@@ -28,7 +28,8 @@ class FrameworkSuite extends BaseFrameworkSuite {
     SmallStackTraceFrameworkSuite,
     AssertionsFrameworkSuite,
     Issue179FrameworkSuite,
-    ScalaCheckExceptionFrameworkSuite
+    ScalaCheckExceptionFrameworkSuite,
+    BoxedFrameworkSuite
   )
   tests.foreach { t => check(t) }
 }

